rpm/containers-common.spec

```diff
@@ -195,13 +199,12 @@ ln -s ../../../..%{_sysconfdir}/yum.repos.d/redhat.repo %{buildroot}%{_datadir}/
 %dir %{_prefix}/lib/containers/storage/overlay-layers
 %{_prefix}/lib/containers/storage/overlay-images/images.lock
 %{_prefix}/lib/containers/storage/overlay-layers/layers.lock
-
 %config(noreplace) %{_sysconfdir}/containers/policy.json
 %config(noreplace) %{_sysconfdir}/containers/registries.conf
 %config(noreplace) %{_sysconfdir}/containers/registries.conf.d/000-shortnames.conf
-%if 0%{?fedora} || 0%{?centos}
-%{_sysconfdir}/pki/rpm-gpg/RPM-GPG-KEY-redhat-release
-%endif
+# This file is owned by redhat-release on RHEL and early testing on RHEL will
+# need packages built on CentOS Stream so it's best ghosted
+%ghost %{_sysconfdir}/pki/rpm-gpg/RPM-GPG-KEY-redhat-release
```

Comment:
So what installs the file now on fedora/centos? 

Why are we shipping this file ourselves. If there is a package for RHEL then why do we not have a separate package on fedora/centos as well? Seems strange that we have to manage the key.

Reply:
Right but on RHEL the keys are not shipped by us. So why are we shipping them on fedora then? Did we ever ask the the owner of the package in RHEL to also put it into fedora? IMO it is strange that containers-common owns these files in fedora...

But well I guess that is a long standing thing and not really related to the issue/fix at hand.

Reply:
Can't remember exactly from the top of my head but maybe one of the reasons why we ship it might be to allow Fedora to use images from RH registries and to understand shortnames pointing there?

Reply:
This particular file is provided by the `redhat-release` package on RHEL (mentioned in the comment) and does not exist on Fedora and CentOS Stream.

@rhatdan can correct me, but IIRC, this file is installed by the containers-common rpm for the case of fedora environments being able to use RHEL containers along with access to RHEL subscription-manager repos.

The Fedora package `distribution-gpg-keys` provides a bunch of keys installed to `/usr/share/distribution-gpg-keys`.

Reply:
Right I get the use case and how this works.
I am questioning why we do this this way? IMO it would make more sense to have the same (redhat-release) package in fedora as well that ships the keys. Then we do not need different packaging logic around this.

The fact the we just pull the key from a web site is not exactly great in my eyes.
https://github.com/containers/common/blob/main/rpm/containers-common.spec#L70-L73

Reply:
ack. This `redhat-release` is one of the few exceptions where things go directly into rhel. Of course, we can ask around to have it in Fedora and CentOS Stream, but I don't really have high hopes it'll happen.

Reply:
Yup this was done way back originally for Docker package, now moby. Moving it to a redhat-release package would be fine with me. But I don't see it as a huge deal either way.